share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml

```diff
@@ -89,6 +93,39 @@ default:
         AWS_ACCESS_KEY_ID: ${PR_MIRRORS_AWS_ACCESS_KEY_ID}
         AWS_SECRET_ACCESS_KEY: ${PR_MIRRORS_AWS_SECRET_ACCESS_KEY}
 
+.generate-base-windows:
+  stage: generate
+  script:
+    - .\share\spack\setup-env.ps1
+    - spack python -m pip install --user boto3 # spack python ensures we're pip installing with Spack's python
+    - spack --version
+    - spack compiler find
+    # These should exist in the MSVC installation
+    - spack external find --not-buildable cmake ninja win-wdk win-sdk wgl msmpi
+    - spack env activate --without-view .\share\spack\gitlab\cloud_pipelines\stacks\$env:SPACK_CI_STACK_NAME
+    - $SPACK_CI_CONFIG_ROOT = ".\share\spack\gitlab\cloud_pipelines\configs"
+    - spack
+      --env-dir .\share\spack\gitlab\cloud_pipelines\stacks\$env:SPACK_CI_STACK_NAME
```

Comment:
I don't think I get why the `--env-dir` arg is needed here. Is the explicit activation of the same environment, above on line 105, not enough?

Reply:
Not until #37951 lands. Right now env support uses `set` to set the ENV variable that Spack looks for to recognize the environments, and Powershell does not support the use of `set` in that manner, so it's necessary to point spack to the env location explicitly.